pkg/controller/dnsrecord/actuator.go

```diff
@@ -51,11 +51,12 @@ func NewActuator(mgr manager.Manager, awsClientFactory awsclient.Factory) dnsrec
 // Reconcile reconciles the DNSRecord.
 func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, dns *extensionsv1alpha1.DNSRecord, _ *extensionscontroller.Cluster) error {
 	// Create AWS client
-	credentials, err := aws.GetCredentialsFromSecretRef(ctx, a.client, dns.Spec.SecretRef, true)
+	authConfig, err := aws.GetCredentialsFromSecretRef(ctx, a.client, dns.Spec.SecretRef, true)
 	if err != nil {
 		return util.DetermineError(fmt.Errorf("could not get AWS credentials: %+v", err), helper.KnownCodes)
 	}
-	awsClient, err := a.awsClientFactory.NewClient(string(credentials.AccessKeyID), string(credentials.SecretAccessKey), getRegion(dns, credentials))
+	authConfig.Region = getRegion(dns, authConfig)
```

Comment:
You also see it here, like the client construction seems a little bit backwards I would say.